server/service/src/number.rs

```diff
@@ -8,9 +11,74 @@ pub fn next_number(
     // Should be done in transaction
     let next_number = connection.transaction_sync(|connection_tx| {
         let repo = NumberRowRepository::new(&connection_tx);
-        let next_number = repo.get_next_number_for_type_and_store(r#type, store_id)?;
-
-        Ok(next_number.number)
+        let number = repo.find_one_by_type_and_store(r#type, store_id)?;
+
+        if number.is_some() {
+            let next_number = repo.get_next_number_for_type_and_store(r#type, store_id, None)?;
+            Ok(next_number.number)
+        } else {
+            match r#type {
+                NumberRowType::InboundShipment => {
+                    let find_number = InvoiceRowRepository::new(&connection_tx)
+                        .find_max_invoice_number(InvoiceRowType::InboundShipment, store_id)?
+                        .map(|n| n + 1);
+                    let next_number =
```

Comment:
I think we should return `found_number` from `match` (do early return for program only). And then just have one `let next_number = repo.get_next_number_for_type_and_store(r#type, store_id, `found_number`)?;`